enhancements/olm/catalogd-console-api.md

```diff
@@ -0,0 +1,670 @@
+
+Consider including folks that also work outside your immediate sub-project.
+-->
+
+### Drawbacks
+
+#### Caching 
+Given the highly-variable nature of responses based on unbound combinations of schema/name/package, 
+any server-side caching will be problematic if not completely impractical. Read: computationally and/or 
+storage expensive.
+
+For example, response for `metas?schema=olm.package&name=foo` and `metas?schema=olm.package&name=bar`
+represent disjoint sets, and the server would have to either ignore `If-Modified-Since` directives 
+in client requests or rely on their veracity uncritically.
+
+Further, if the underlying implementation relies on anything other than a dictionary of cached requests 
+(&parameters) to dictate date-relevant responses (for e.g. relying on the catalog unpack time), 
+then any server-side caching would be erroneous.  
+
+The existing `all` endpoint also incentivizes clients to conserve resources via local cache to avoid making 
+many (potentially duplicate) requests.  However, the OCP console proof of concept 
+required what was deemed an unsupportable amount of code and complexity to cache, decompose, and render the 
+complete FBC.
+
+#### Complexity
+Server implementation is much more complex and resource intensive due to the need to index catalog 
+content and serve variable requests.
+
+#### Completeness
+The previous `all` endpoint always returns valid FBC.  The new service cannot make that promise, 
+so clients could make incorrect assumptions about the suitability of results.  See Open Questions.
+
+
+<!--
+The idea is to find the best form of an argument why this enhancement should
+_not_ be implemented.
+
+What trade-offs (technical/efficiency cost, user experience, flexibility,
+supportability, etc) must be made in order to implement this? What are the reasons
+we might not want to undertake this proposal, and how do we overcome them?
+
+Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+aligned with existing user expectations?  Will it be a significant maintenance
+burden?  Is it likely to be superceded by something else in the near future?
+-->
+
+## Open Questions [optional]
+<!--
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance,
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this?
+-->
+ > 1. If a query comes in with `/api/v1/metas?package=foo`, should we include the blob with schema: `olm.package` and name: `foo`?
+
+We feel that it is incorrect for the metas service endpoint to mutate the data model (specifically, to create a synthetic package attribute for the `olm.package` schema).  To access all the data modeled for an installable package, separate queries need to be made for the package-level metadata (`schema=olm.package&name=foo`) versus the channel/bundle-level metadata (`package=foo`).
```

Comment:
This section could probably use a little more explanation as to why this is even a question, ie include the context that `olm.package` objects do not contain `packageName`